src/main/java/net/dv8tion/jda/api/utils/FileProxy.java

```diff
@@ -0,0 +1,334 @@
+/*
+ * Copyright 2015 Austin Keener, Michael Ritter, Florian Spieß, and the JDA contributors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package net.dv8tion.jda.api.utils;
+
+import net.dv8tion.jda.api.exceptions.HttpException;
+import net.dv8tion.jda.internal.requests.FunctionalCallback;
+import net.dv8tion.jda.internal.requests.Requester;
+import net.dv8tion.jda.internal.utils.Checks;
+import net.dv8tion.jda.internal.utils.FutureUtil;
+import net.dv8tion.jda.internal.utils.IOUtil;
+import okhttp3.Call;
+import okhttp3.HttpUrl;
+import okhttp3.OkHttpClient;
+import okhttp3.Request;
+
+import javax.annotation.Nonnull;
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.UncheckedIOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.nio.file.StandardCopyOption;
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+import java.util.function.Function;
+
+/**
+ * A utility class to download files.
+ */
+public class FileProxy
+{
+    private static volatile OkHttpClient defaultHttpClient;
+
+    private final String url;
+    private OkHttpClient customHttpClient;
+
+    /**
+     * Constructs a new {@link FileProxy} for the provided URL.
+     *
+     * @param url
+     *        The URL to download from
+     *
+     * @throws IllegalArgumentException
+     *         If the provided URL is null
+     */
+    public FileProxy(@Nonnull String url)
+    {
+        Checks.notNull(url, "URL");
+        this.url = url;
+    }
+
+    /**
+     * Sets the default OkHttpClient used by {@link FileProxy} and {@link ImageProxy}.
+     * <br>This can still be overridden on a per-instance basis with {@link #withClient(OkHttpClient)}.
+     *
+     * @param  httpClient
+     *         The default {@link OkHttpClient} to use while making HTTP requests
+     *
+     * @throws IllegalArgumentException
+     *         If the provided {@link OkHttpClient} is null
+     */
+    public static void setHttpClient(@Nonnull OkHttpClient httpClient)
```

Comment:
setDefaultHttpClient makes more sense IMHO

otherwise people might confuse it that it sets the client of that instance specifically